src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -150,50 +177,73 @@ public string GetServerUrl(CommandSettings command)
-            Trace.Info($"PoolId for deployment group '{deploymentGroupName}' is '{poolId}'.");
-            
-            return poolId;
+            var deploymentGroup =  await GetDeploymentGroupAsync(_projectName, deploymentGroupName);
+            Trace.Info($"PoolId for deployment group '{deploymentGroupName}' is '{deploymentGroup.Pool.Id}'.");
+            Trace.Info($"Project id for deployment group '{deploymentGroupName}' is '{deploymentGroup.Project.Id.ToString()}'.");
+
+            return new AgentSettings() { PoolId = deploymentGroup.Pool.Id, DeploymentGroupId = deploymentGroup.Id, ProjectId = deploymentGroup.Project.Id.ToString()};
         }
 
         public string GetFailedToFindPoolErrorString() => StringUtil.Loc("FailedToFindDeploymentGroup");
 
-        public async Task<TaskAgent> UpdateAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public string GetAgentWithSameNameAlreadyExistErrorString(AgentSettings agentSettings)
+        {
+            return StringUtil.Loc("DeploymentMachineWithSameNameAlreadyExistInDeploymentGroup", agentSettings.DeploymentGroupId, agentSettings.AgentName);
+        }
+
+        public async Task<TaskAgent> UpdateAgentAsync(AgentSettings agentSettings, TaskAgent agent, CommandSettings command)
         {
-            agent = await _agentServer.UpdateAgentAsync(poolId, agent);
-            await GetAndAddTags(agent, command);
+            var deploymentMachine = new DeploymentMachine() { Agent = agent };
+            deploymentMachine = await _deploymentGroupServer.ReplaceDeploymentMachineAsync(new Guid(agentSettings.ProjectId), agentSettings.DeploymentGroupId, agent.Id, deploymentMachine);
 
-            return agent;
+            await GetAndAddTags(deploymentMachine.Id, agentSettings, agent, command);
+            return deploymentMachine.Agent;
         }
 
-        public async Task<TaskAgent> AddAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public async Task<TaskAgent> AddAgentAsync(AgentSettings agentSettings, TaskAgent agent, CommandSettings command)
         {
-            agent = await _agentServer.AddAgentAsync(poolId, agent);
-            await GetAndAddTags(agent, command);
+            var deploymentMachine = new DeploymentMachine(){ Agent = agent };
+            deploymentMachine = await _deploymentGroupServer.AddDeploymentMachineAsync(new Guid(agentSettings.ProjectId), agentSettings.DeploymentGroupId, deploymentMachine);
+            
+            await GetAndAddTags(deploymentMachine.Id, agentSettings, deploymentMachine.Agent, command);
 
-            return agent;
+            return deploymentMachine.Agent;
         }
 
-        public Task DeleteAgentAsync(int agentPoolId, int agentId)
+        public async Task DeleteAgentAsync(AgentSettings agentSettings)
         {
-            return _agentServer.DeleteAgentAsync(agentPoolId, agentId);
+            string currentAction = StringUtil.Loc("UninstallingService");
```

Comment:
again, i wouldn't expect anytime i can DeleteAgent should print out uninstall service.